### ISSUE TITLE
Option to allow reassigning default parameters when no-param-reassign is set

### ISSUE THREAD

**Issue by zviwex** [eslint/eslint#14189]
I have a suggestion to implement a new Option to the "no-param-reassign" rule:
The option will be called "ignoreDefaultParameters" and will cause the role to ignore function default parameters and let them be reassigned.

The motivation here is that usually, one of default parameters' behaviors is to edit them if it's not set.

**What rule do you want to change?**
no-param-reassign

**Does this change cause the rule to produce more or fewer warnings?**
Subjected to configuration, fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**
```js
/*eslint no-param-reassign: ["error", { "ignoreDefaultParameters": true}]*/

function foo(bar = 1) {
  if(bar != 1) { 
    bar = 2;
  }
}
```

**What does the rule currently do for this code?**
The rule has no support for this option.
**What will the rule do after it's changed?**
By default, nothing changes. When the option introduces, the default parameters are ignored.

**Comment by mdjermanovic** [comment]
Hi @zviwex, thanks for the issue!

> The motivation here is that usually, one of default parameters' behaviors is to edit them if it's not set.

Can you clarify this? I believe it's actually less common to reassign parameters that have default values than those without.

> ```js
> /*eslint no-param-reassign: ["error", { "ignoreDefaultParameters": true}]*/
> 
> function foo(bar = 1) {
>   if(bar != 1) { 
>     bar = 2;
>   }
> }
> ```

I'm not sure that I understand this example - it looks like the `bar` argument value isn't used at all?

**Comment by zviwex** [comment]
More reliable example:
```js
function getItem(item = '') {
  if(item === '') { 
    item = getDefaultItem();
  }
 return db.getItem(item);
}
```

**Comment by mdjermanovic** [comment]
Thanks for the additional example!

I don't think this is common enough to warrant an option that disables checking all parameters with default values, and it looks to me like a good case for [`eslint-disable-*`](https://eslint.org/docs/user-guide/configuring/rules#using-configuration-comments-1) comments.
